debug_module.tex

```diff
@@ -98,7 +98,8 @@ \section{Selecting Harts} \label{selectingharts}
 
 A debugger can enumerate all the harts
 attached to the DM by selecting each hart starting from 0
-until \Fanynonexistent in \Rdmstatus is 1.
+until either \Fanynonexistent in \Rdmstatus is 1, or until the value it reads
+back from \Fhartsel does not match what it wrote there.
```

Comment:
It sounds like you're considering letting people tie eg. bits 1 and 9:3 to 0, leaving as the only valid values 0, 1, 4, 5. An implementation could trivially rewire the bits so that bits 9:2 are tied to 0 instead, which makes the spec simpler as well as the debugger implementation.

Did I misinterpret what you said?

Reply:
That was what my comment was saying... I think your change made it more clear that it must be the LSB which are wired.

Reply:
I think I see what you're saying.

If we allow tieing arbitrary bits to 0, then we can add more range to hartsel by expanding the register on the LSB side (where we have space), and in old implementations those are simply detected as tied to 0, since that's what the spec says. Is that right?

Reply:
I'm not advocating changing the location of hartsel[0] in the register, even if hartsel[10] ends up at a lower location in DMCONTROL register